tracer/src/Datadog.Trace/Datadog.Trace.csproj

```diff
@@ -49,6 +49,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Fody" Version="6.8.0">
```

Comment:
We have to be very careful with new assembly references because if we statically reference anything not in the GAC then we risk instrumentation errors for .NET Framework applications. 

Running the unit test `Datadog.Trace.Tests.PublicApiTests.DatadogTraceTests.AssemblyReferencesHaveNotChanged` should generate the following `*.received.txt` snapshot files, which we would persist in the repo by updating the corresponding `*.verified.txt` if we deem them safe to add:
- `tracer\test\Datadog.Trace.Tests\Snapshots\PublicApiTests.Datadog.Trace.AssemblyReferencesHaveNotChanged.net6.0.verified.txt`
- `tracer\test\Datadog.Trace.Tests\Snapshots\PublicApiTests.Datadog.Trace.AssemblyReferencesHaveNotChanged.net461.verified.txt`
- `tracer\test\Datadog.Trace.Tests\Snapshots\PublicApiTests.Datadog.Trace.AssemblyReferencesHaveNotChanged.netcoreapp3.1.verified.txt`
- `tracer\test\Datadog.Trace.Tests\Snapshots\PublicApiTests.Datadog.Trace.AssemblyReferencesHaveNotChanged.netstandard2.0.verified.txt`

Reply:
Note that while indeed the `AssmeblyReferencesHaveNotChanged` test is currently failing on some other assemblies and @dudikeleti is still investigating, is it not failing on this reference, because Fody itself is not referenced by `Datadog.Trace.dll` as a result of this line of code. We are using Fody strictly as intended, as a bytecode weaver, so the only affect of adding this reference is that we slightly manipulate the bytecode of `Datadog.Trace.dll` as a post-compilation step. 

While doing this bytecode weaving adds a bit of complexity, we feel this is justified because it is required in order to allow us to leverage the very nice performance improvements we are able to gain by leveraging `System.Memory` in both .NET Framework and .NET Core. The performance gains of doing the symbol extraction this way are huge (@dudikeleti can share the benchmarks if you're interested), but I'd argue the performance gains we can obtain by having access to `System.Memory` as a vendored dependency in future improvements across the client libraries are equally promising.

Reply:
Sounds good! I'll take a look at the updated `AssmeblyReferencesHaveNotChanged` snapshots with the changed assembly references and leave any feedback there